source/_docs/third-party-cdns.md

```diff
@@ -0,0 +1,103 @@
+---
+title: Using a Third Party CDN
+description: Understand required configurations to stack a third-party CDN service on top of Pantheon's Global CDN.
+tags: [cacheedge]
+---
+Most projects do not need a third-party CDN service, as Pantheon offers a [Global CDN](/docs/global-cdn/) for all sites out of the box with the following benefits:
+
+* It includes [free and automatic HTTPS service](/docs/https)
+* It is heavily optimized for website performance
+* It is configured, maintained, and supported by Pantheon
+* It is available at no additional cost
+
+However, if there is some custom feature the project requires from another service provider (e.g., WAF, custom certificates, page rules, etc.) you can stack the service on top of Pantheon's Global CDN using the following configurations.
+
+<table class="table  table-bordered table-responsive">
+  <thead>
+    <tr>
+      <th>Service Provider</th>
+      <th>Known to work with Pantheon</th>
+      <th>Configuration Notes</th>
+    </tr>
+  </thead>
+  <tbody>
+    <tr>
+      <th>Cloudflare</th>
+      <td>✓</td>
+      <td><a href="#cloudflare">See more</a></td>
+    </tr>
+    <tr>
+      <th>Fastly</th>
+      <td>✓</td>
+      <td><a href="#fastly">See more</a></td>
+    </tr>
+    <tr>
+      <th>AWS CloudFront</th>
+      <td>✓</td>
+      <td><a href="#aws-cloudfront">See more</a></td>
+    </tr>
+    <tr>
+      <th>Verizon EdgeCast</th>
+      <td>✓</td>
+      <td></td>
+    </tr>
+  </tbody>
+</table>
+
+In general, you must assume HTTPS in the application (within WordPress or Drupal) in addition to requiring HTTPS within configurations for your third party CDN service provider. Otherwise, the site will encounter routing issues and possible failures.
+
+
+## Cloudflare
+1. Assume HTTPS within WordPress or Drupal. For details, see [Switching Sites from HTTP to HTTPS](/docs/http-to-https/).
+2. Select **Crypto** from the Cloudflare menu bar and set SSL mode to **Full** (or potentially Full, Strict), but not Flexible.
+  ![Enable SSL](/source/docs/assets/images/cloudflare-ssl.png)
+3. Scroll down and enable **Always use HTTPS**
+  ![Cloudflare Always HTTPS](/source/docs/assets/images/cloudflare-always-https.png)
+4. Scroll down and enable **Automatic HTTPS Rewrites**
+5. Remove existing redirects configured via PHP in `settings.php` or `wp-config.php`.
+6. Proceed with DNS configuration as describe in Option 1, but make sure the cloud is toggled orange, not gray.
+
+## Fastly
+<div class="alert alert-info">
+  <h4 class="info">Note</h4>
```

Comment:
Let's remove this note. Would be excellent if all live sites on Pantheon have the custom domain added to site dashboard for better supportability.

Reply:
@davidstrauss ^^ That sound reasonable?

Reply:
Sounds reasonable, though there are two downsides I can think of:

1. We'll provision entries on a cert that will go unused.
2. It makes it easier for the site owner to misconfigure something to rely on the public domain having a cert entry on Global CDN. This could cause downtime if the CDN stacking breaks renewal.

Reply:
@ari-gold thoughts?

Reply:
Also, even if they don't rely on us provisioning their public domains onto Let's Encrypt certs, it could cause a high error rate as we attempt to do so through their CDN.